docs/src/props-table.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import {Box, Link, Label} from '@primer/components'
+import Table from '@primer/gatsby-theme-doctocat/src/components/table'
+
+function PropsTable({children}) {
+  return (
+    <Table>
+      <thead>
+        <tr>
+          <Box as="th" textAlign="left">
+            Name
+          </Box>
+          <Box as="th" textAlign="left">
+            Type
+          </Box>
+          <Box as="th" textAlign="left">
+            Default
+          </Box>
+          <Box as="th" textAlign="left">
+            Description
+          </Box>
+        </tr>
+      </thead>
+      <tbody>{children}</tbody>
+    </Table>
+  )
+}
+
+function Row({name, type, defaultValue, description, required}) {
+  return (
+    <tr>
+      <Box as="td" fontFamily="mono" fontSize={1} fontWeight="bold" sx={{whiteSpace: 'nowrap'}}>
+        {typeof name === 'function' ? name() : name}
+        {required ? (
+          <>
+            {` `}
+            <Label style={{verticalAlign: 'text-top'}}>Required</Label>
+          </>
+        ) : null}
+      </Box>
+      <Box as="td" fontFamily="mono" fontSize={1}>
+        {typeof type === 'function' ? type() : type}
+      </Box>
+      <Box as="td" fontFamily="mono" fontSize={1}>
+        {typeof defaultValue === 'function' ? defaultValue() : defaultValue}
+      </Box>
+      <td>{typeof description === 'function' ? description() : description}</td>
+    </tr>
+  )
+}
+
+function AsRow({defaultValue}) {
+  return (
+    <Row
+      name="as"
+      defaultValue={defaultValue}
+      type="ReactElement"
+      description="The underlying element to render — either a DOM element or a React component."
```

Comment:
Should we add a note here about also accepting all props associated with the value of the as prop?

Reply:
Since sometimes we'll accept DOM props but _not_ an `as` prop, I've started another helper row that explains DOM props, and additionally it will point out that its value changes according to the `as` prop. WDYT?